src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -191,99 +188,49 @@ private JsonObject extractUserFromLookupResponse(HttpResponse<Buffer> res, Strin
-            }
-            if(token == null) {
-              promise.fail(String.format("Got response %s fetching token, but content is null",
-                  response.statusCode()));
-            } else {
-              logger.debug("Got token " + token + " from authz");
-              promise.complete(token);
-            }
-          } else {
-            promise.fail("Got response " + response.statusCode() + " fetching token");
-          }
-        } catch(Exception e) {
-          promise.fail(String.format("Error getting token: %s", e.getLocalizedMessage()));
-        }
+    return request.sendJson(new JsonObject().put("payload", payload)).map(response -> {
+      if (response.statusCode() != 200 && response.statusCode() != 201) {
+        throw new RuntimeException("Got response " + response.statusCode() + " fetching token");
       }
+      String token = response.statusCode() == 200
+        ? response.getHeader(OKAPI_TOKEN_HEADER)
+        : response.bodyAsJsonObject().getString("token");
+      if (token == null) {
+        throw new RuntimeException(String.format("Got response %s fetching token, but content is null",
+          response.statusCode()));
+      }
+      logger.debug("Got token " + token + " from authz");
+      return token;
     });
-    return promise.future();
   }
 
   private Future<String> fetchRefreshToken(String userId, String sub, String tenant,
-      String okapiURL, String requestToken) {
-    Promise<String> promise = Promise.promise();
+    String okapiURL, String requestToken) {
     HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + "/refreshtoken");
     request.putHeader(OKAPI_TENANT_HEADER, tenant)
-      .putHeader(OKAPI_TOKEN_HEADER, requestToken)
-      .putHeader(CONTENT_TYPE, APPLICATION_JSON_CONTENT_TYPE)
-      .putHeader(ACCEPT, APPLICATION_JSON_CONTENT_TYPE);
+      .putHeader(OKAPI_TOKEN_HEADER, requestToken);
 
     JsonObject payload = new JsonObject().put("userId", userId).put("sub", sub);
-    request.sendJsonObject(payload, ar -> {
-      if(ar.failed()) {
-        promise.fail(ar.cause());
-      } else {
-        HttpResponse<Buffer> response = ar.result();
-        if(response.statusCode() != 201) {
-          String message = String.format("Expected code 201 from /refreshtoken, got %s",
-              response.statusCode());
-          promise.fail(message);
-        } else {
-          String refreshToken;
-          try {
-            refreshToken = response.bodyAsJsonObject().getString("refreshToken");
-          } catch(Exception e) {
-            promise.fail(e);
-            return;
-          }
-          promise.complete(refreshToken);
-        }
+    return request.sendJsonObject(payload).map(response -> {
+      if (response.statusCode() != 201) {
+        String message = String.format("Expected code 201 from /refreshtoken, got %s",
+          response.statusCode());
+        throw new RuntimeException(message);
+
```

Comment:
I would use
https://vertx.io/docs/vertx-web-client/java/#response-predicates
https://vertx.io/docs/apidocs/io/vertx/ext/web/client/predicate/ResponsePredicate.html
to check the HTTP status code.

Reply:
Done for some cases